lib/coinbase/stake.rb

```diff
@@ -0,0 +1,161 @@
+      # @return [String] An unsigned transaction used to claim_stake.
+      def build_claim_stake_operation(protocol, network, params, opts = { mode: :partial })
+        body = {}
+
+        case protocol_network(protocol, network)
+        when 'ethereum-holesky'
+          return if missing_required_params?(params, [:address])
+          return if not_partial?(opts)
+
+          body = {
+            'action' => action("#{protocol}_kiln".to_sym, network, :claim_stake),
+            'ethereum_kiln_staking_parameters' => {
+              'claim_stake_parameters' => {
+                'staker_address' => (params[:address]).to_s
+              }
+            }
+          }
+        end
+
+        call_client(body)
+      end
+
+      # Checks to see if the input parameters have all required fields.
+      # @param params [Hash] User provided parameters for a staking operation.
+      # @param required_params [Hash] The required parameters needed to create a staking operation.
+      # @return [Boolean] If the input parameters have all the required parameters populated.
+      def missing_required_params?(params, required_params)
+        raise ArgumentError, "missing required params #{required_params}" unless required_params.all? do |key|
+          params.key?(key)
+        end
+      end
+
+      # Creates an action resource name for the body of a staking operation.
+      # @param protocol [Symbol] The protocol name.
+      # @param network [Symbol] The network name.
+      # @param action [Symbol] The action which translates to a specific staking operation.
+      # @return [String] The action resource name.
+      def action(protocol, network, action)
+        "protocols/#{protocol}/networks/#{network}/actions/#{action}"
+      end
+
+      # Creates a protocol-network pair.
+      # @param protocol [Symbol] The protocol name.
+      # @param network [Symbol] The network name.
+      # @return [String] The protocol-network combo.
+      def protocol_network(protocol, network)
+        "#{protocol}-#{network}"
+      end
+
+      # Checks if the mode is set to partial.
+      # @param opts [Hash] The options passed in by the user.
+      # @return [Boolean] If the mode is partial.
+      def not_partial?(opts)
+        raise ArgumentError, "required mode to be 'partial'" if opts[:mode] != :partial
+      end
+
+      # Sets up the options and calls the Staking API service backend.
+      # @param body [Hash] Holds the different options passed into the API call.
+      # @return [String] The unsigned transaction from the response body.
+      def call_client(body)
+        opts = {
+          header_params: {
+            'Content-Type' => 'application/json',
+            'Accept' => 'application/json'
+          },
+          query_params: {},
+          return_type: 'Object'
+        }
+
+        call_opts = opts.merge(body: body)
+
+        response = Coinbase.call_api do
+          staking_api.call_api(:POST, '/v1/workflows', call_opts)
```

Comment:
I think these should be routed through CDPService. CC @alex-stone

Reply:
Yeah, the first feature phase can go through the existing staking service, but this will go through CDP service before it lands in a release!